File: day2/main2.py
rock = 0 # +1 A
paper = 0 # +2 B
scissors = 0 # +3 C
# Lose = X
# Draw = Y 
# Win = Z
win = 0 # +6
tie = 0 # +3

file = open("file.txt")
for line in file:
    line = line.rstrip().split(" ")
    # print(line)
    if line[0] == "A":
        if line[1] == "X":
            scissors +=1
        if line[1] == "Y":
            rock +=1
            tie +=1
        if line[1] == "Z":
            paper +=1
            win +=1

    if line[0] == "B":
        if line[1] == "X":
            rock+=1
        if line[1] == "Y":
            paper +=1
            tie +=1
        if line[1] == "Z":
            scissors +=1
            win +=1

    if line[0] == "C":
        if line[1] == "X":
            paper +=1
        if line[1] == "Y":
            scissors +=1
            tie +=1
        if line[1] == "Z":
            win +=1
            rock +=1

score = (rock) + (paper*2) + (scissors*3) + (tie*3) + (win*6)

print(score)

File: C++/day4/main2.cpp
#include <iostream>
#include <fstream>
#include <string>

// a = rock   b = paper   c = scissors

// x = rock  y = paper z = scissor

// rock = 1   paper = 2   scissors = 3
int main() {

std::ifstream file("file.txt");
std::string   line;

int           rock     = 0;
int           paper    = 0;
int           scissors = 0;
int           win      = 0;
int           tie      = 0;

while ( file >> line) {


  if( line == "A" ) {
    file >> line;
    if( line == "X") {
      scissors++;
    }
    if( line == "Y") {
      rock++;
      tie++;
    }
    if( line == "Z") {
      paper++;
      win++;
    }
  }

  if( line == "B" ) {
    file >> line;
    if ( line == "X") {
      rock++;
    }
    if( line == "Y") {
      paper++;
      tie++;
    }
    if( line == "Z") {
      scissors++;
      win++;
    }
  }

    if( line == "C" ) {
      file >> line;
      if( line == "X" )
      {
        paper++;
      }
      if( line == "Y" )
      {
        scissors++;
        tie++;
      }
      if( line == "Z" )
      {
        rock++;
        win++;
      }
    }

}
int score = ( rock * 1 ) + ( paper * 2 ) + ( scissors * 3 ) + (win* 6) + (tie*3);

std::cout << score << std::endl;

return 0;


// x = lose
// y = tie
// z = win
}
